pkg/helm/actions/get_repos.go

```diff
@@ -21,119 +25,192 @@ var (
 		Version:  "v1beta1",
 		Resource: "helmchartrepositories",
 	}
+
+	configNamespace = "openshift-config"
 )
 
-func prepareAnnotations(url, name string) map[string]string {
-	return map[string]string{
-		"chart_repo_url": url,
-		"repo_name":      name,
+type HttpClientCertGetter interface {
+	CaCertificates() (*tls.Config, error)
+	TlsCertificates() (*tls.Config, error)
+}
+
+type ConfigMapReader interface {
+	ConfigMapValue(key string) ([]byte, error)
+	SecretValue(key string) ([]byte, error)
+}
+
+type HelmConfigBuilder struct {
```

Comment:
this should actually something like `HelmChartRepoConfigGetter`, and interface:

```
type HelmChartRepoGetter interface {
  List() []HelmChartRepo
}

```

and then each `HelmCharRepo` could implement such methods like `IndexFile()`, etc.